sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/coders/AwsCoders.java

```diff
@@ -0,0 +1,169 @@
+
+import com.amazonaws.ResponseMetadata;
+import com.amazonaws.http.HttpResponse;
+import com.amazonaws.http.SdkHttpMetadata;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.coders.MapCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.coders.VarIntCoder;
+import org.apache.beam.sdk.util.VarInt;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableMap;
+
+public final class AwsCoders {
+
+  private AwsCoders() {}
+
+  public static Coder<ResponseMetadata> responseMetadata() {
+    return ResponseMetadataCoder.of();
+  }
+
+  public static Coder<SdkHttpMetadata> fullSdkHttpMetadata() {
+    return FullSdkHttpMetadataCoder.of();
+  }
+
+  public static Coder<SdkHttpMetadata> minimalSdkHttpMetadata() {
+    return MinimalSdkHttpMetadataCoder.of();
+  }
+
+  private static class ResponseMetadataCoder extends Coder<ResponseMetadata>
+      implements Serializable {
+
+    private static final ResponseMetadataCoder INSTANCE = new ResponseMetadataCoder();
+
+    private ResponseMetadataCoder() {}
+
+    public static ResponseMetadataCoder of() {
+      return INSTANCE;
+    }
+
+    @Override
+    public void encode(ResponseMetadata value, OutputStream outStream)
+        throws CoderException, IOException {
+      MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())
+          .encode(
+              ImmutableMap.of(
+                  ResponseMetadata.AWS_REQUEST_ID, value != null ? value.getRequestId() : ""),
+              outStream);
+    }
+
+    @Override
+    public ResponseMetadata decode(InputStream inStream) throws CoderException, IOException {
+      return new ResponseMetadata(
+          MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()).decode(inStream));
+    }
+
+    @Override
+    public List<? extends Coder<?>> getCoderArguments() {
+      return ImmutableList.of();
+    }
+
+    @Override
+    public void verifyDeterministic() throws NonDeterministicException {
+      MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()).verifyDeterministic();
+    }
+  }
+
+  private static class FullSdkHttpMetadataCoder extends Coder<SdkHttpMetadata>
+      implements Serializable {
+
+    private static final FullSdkHttpMetadataCoder INSTANCE = new FullSdkHttpMetadataCoder();
+
+    private FullSdkHttpMetadataCoder() {}
+
+    public static FullSdkHttpMetadataCoder of() {
+      return INSTANCE;
+    }
+
+    @Override
+    public void encode(SdkHttpMetadata value, OutputStream outStream)
+        throws CoderException, IOException {
+      VarInt.encode(value != null ? value.getHttpStatusCode() : 0, outStream);
```

Comment:
Same suggestion for wrapping NullableCoder within `fullSdkHttpMetadata` and not handling the null case here.